source/agora/common/EnrollmentManager.d

```diff
@@ -290,6 +335,48 @@ public class EnrollmentManager
 
         return false;
     }
+
+    /***************************************************************************
+
+        Logs message
+
+        Params:
+            msg = the log message to be logged
+            enroll = the Enrollment object, the information of which will be logged
+
+    ***************************************************************************/
+
+    private void logMessage(string msg, const ref Enrollment enroll) @safe nothrow
```

Comment:
Your goal is to have a `nothrow` log function, right ? From just looking at the function, it is not obvious.

Something like:
```
This function is a helper to deal with `log.info` not being `nothrow`, which forces to add a `try` / `catch` block when logging an exception from a `catch` block.
```

Also, you could merge those two by simply making `Exception` optional and default to `null` (which is possible if you remove `ref`).

Reply:
You're right. The function is created to log messages without a `try` / `catch` block. And I will merge the two functions I created.